### PR TITLE
feat: table configuration with drag and drop 

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: #4879 

### Description:

Note: Only introduced a couple of simple mantine components to avoid a big refactor and make it really different from the rest of the chart config.

Without pivot:
Captura de ecrã 2023-05-25, às 17 41 29

With pivot:
Captura de ecrã 2023-05-25, às 17 58 57

### CODE REVIEW

**Comment by TuringLovesDeathMetal** [comment]
Okay, just played around with this a bit and was really confused why I couldn't find metrics as rows. 

Can we have this option appear, but greyed out/not able to interact with it? And on-hover it says `to use metrics as rows, you need to pivot your table`

Then once you have at least one dimension pivoted, it appears as usable

image

**Comment by TuringLovesDeathMetal** [comment]
@ZeRego sorry, one tiny change to copy. 

Can it say:

`To use metrics as rows, you need to add a pivot.`

**Comment by TuringLovesDeathMetal** [comment]
And can we remove the row totals being greyed out? 

image


I don't think we need to show them until they're available

**Comment by joaoviana** [comment]
https://github.com/lightdash/lightdash/assets/7611706/83989e82-80d9-4fba-937d-0e25ef6dd60b

not sure if part of the PR scope, but I can't reorder `rows`. @ZeRego is that ok?

File: packages/frontend/src/components/TableConfigPanel/ColumnConfiguration.tsx [hunk 1]
```diff
@@ -9,13 +9,11 @@ import {
 } from '@tabler/icons-react';
 import React from 'react';
 import { useVisualizationContext } from '../LightdashVisualization/VisualizationProvider';
-import { ColumnConfigurationWrapper } from './ColumnConfiguration.styles';
 
-const ColumnConfiguration: React.FC = () => {
+const ColumnConfiguration: React.FC<{ fieldId: string }> = ({ fieldId }) => {
```
Comment (joaoviana) [inline packages/frontend/src/components/TableConfigPanel/ColumnConfiguration.tsx:13]: nice

**Comment by ZeRego** [comment]
> not sure if part of the PR scope, but I can't reorder rows. @ZeRego is that ok?
@joaoviana Yes, it is out of scope.

**Review by joaoviana — APPROVED** [review]

**Review by TuringLovesDeathMetal — APPROVED** [review]
Looks good to me!

**Pull request merged**